### PR TITLE
feat: add C implementation for `math/base/special/acsch`

### LINKED ISSUE #860: [RFC]: Add C implementation for `@stdlib/math/base/special/acsch`
### Description

This RFC proposes adding native C implementation for [`stdlib/math/base/special/acsch`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/@stdlib/math/base/special/acsch).

```c
double stdlib_base_acsch( const double x );
```

### Related Issues

None.

### Questions

No.

### Other

No.

### Checklist

- [X] I have read and understood the [Code of Conduct](https://github.com/stdlib-js/stdlib/blob/develop/CODE_OF_CONDUCT.md).
- [X] Searched for existing issues and pull requests.
- [X] The issue name begins with `RFC:`.

### PR DESCRIPTION
**Pull request by yaswanthkosuru**
Resolves  https://github.com/stdlib-js/stdlib/issues/860

## Description
FEATURE TO
ADD C IMPLEMENTATION FOR ACSCH FUNCTION 

This pull request:

-   resolves https://github.com/stdlib-js/stdlib/issues/860

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

File: lib/node_modules/@stdlib/math/base/special/acsch/README.md [hunk 1]
```diff
@@ -83,6 +83,91 @@ for ( i = 0; i < x.length; i++ ) {
...
+#include "stdlib/math/base/special/acsch.h"
+```
+
+#### stdlib_base_acsch( x )
+
+Computes the [hyperbolic arccosecant][inverse-hyperbolic-functions] of `x`.
+
+```c
+double out = stdlib_base_acsch( 2.0 );
+// returns ~0.5493
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/README.md:118]:
```suggestion
double out = stdlib_base_acsch( 1.0 );
// returns ~0.881
```

File: lib/node_modules/@stdlib/math/base/special/acsch/benchmark/benchmark.native.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/benchmark/benchmark.native.js:4]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/benchmark/c/Makefile [hunk 3]
```diff
@@ -0,0 +1,107 @@
+#/
+# @license Apache-2.0
+#
+# Copyright (c) 2018 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/benchmark/c/Makefile:4]:
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/benchmark/c/benchmark.c [hunk 4]
```diff
@@ -0,0 +1,135 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/benchmark/c/benchmark.c:4]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/benchmark/c/benchmark.c [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+double benchmark() {
+	double elapsed;
+	double x;
+	double y;
+	double t;
+	int i;
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		x = ( 100.0*rand_double() ) + 1.1;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/benchmark/c/benchmark.c:103]: Any specific reason for `1.1` ?

File: lib/node_modules/@stdlib/math/base/special/acsch/benchmark/c/native/Makefile [hunk 6]
```diff
@@ -0,0 +1,146 @@
+#/
+# @license Apache-2.0
+#
+# Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/benchmark/c/native/Makefile:4]:
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/benchmark/c/native/benchmark.c [hunk 7]
```diff
@@ -0,0 +1,136 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/benchmark/c/native/benchmark.c:4]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/benchmark/c/native/benchmark.c [hunk 8]
```diff
@@ -0,0 +1,136 @@
...
+double benchmark() {
+	double elapsed;
+	double x;
+	double y;
+	double t;
+	int i;
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		x = ( 100.0*rand_double() ) - 50.0;
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/benchmark/c/native/benchmark.c:104]: This is better

File: lib/node_modules/@stdlib/math/base/special/acsch/benchmark/python/benchmark.py [hunk 9]
```diff
@@ -0,0 +1,97 @@
+#!/usr/bin/env python
+#
+# @license Apache-2.0
+#
+# Copyright (c) 2018 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/benchmark/python/benchmark.py:5]:
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/benchmark/python/benchmark.py [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+    print_summary(REPEATS, REPEATS)
+
+
+def main():
+    """Run the benchmark."""
+    benchmark()
+
+
+if __name__ == "__main__":
+    main()
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/benchmark/python/benchmark.py:97]: The issue requests to add C implementation so does PR name. You will need to change either name or revert this changes.

File: lib/node_modules/@stdlib/math/base/special/acsch/binding.gyp [hunk 11]
```diff
@@ -0,0 +1,170 @@
+# @license Apache-2.0
+#
+# Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/binding.gyp:3]:
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/examples/c/Makefile [hunk 12]
```diff
@@ -0,0 +1,146 @@
+#/
+# @license Apache-2.0
+#
+# Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/examples/c/Makefile:4]:
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/examples/c/example.c [hunk 13]
```diff
@@ -0,0 +1,31 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/examples/c/example.c:4]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/include.gypi [hunk 14]
```diff
@@ -0,0 +1,54 @@
+# @license Apache-2.0
+#
+# Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/include.gypi:3]:
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/include.gypi [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+      '<!@(node -e "var arr = require(\'@stdlib/utils/library-manifest\')(\'./manifest.json\',{},{\'basedir\':process.cwd(),\'paths\':\'posix\'}).libraries; for ( var i = 0; i < arr.length; i++ ) { console.log( arr[ i ] ); }")',
+    ],
+
+    # Library directories:
+    'library_dirs': [
+      '<!@(node -e "var arr = require(\'@stdlib/utils/library-manifest\')(\'./manifest.json\',{},{\'basedir\':process.cwd(),\'paths\':\'posix\'}).libpath; for ( var i = 0; i < arr.length; i++ ) { console.log( arr[ i ] ); }")',
+    ],
+  }, # end variables
+}
+s
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/include.gypi:54]: This s should not be there, please verify

File: lib/node_modules/@stdlib/math/base/special/acsch/include/stdlib/math/base/special/acsch.h [hunk 16]
```diff
@@ -0,0 +1,38 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/include/stdlib/math/base/special/acsch.h:4]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/include/stdlib/math/base/special/acsch.h [hunk 17]
```diff
@@ -0,0 +1,38 @@
...
+
+/*
+* If C++, prevent name mangling so that the compiler emits a binary file having undecorated names, thus mirroring the behavior of a C compiler.
+*/
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+* Computes the inverse hyperbolic cotangent of a double-precision floating-point number.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/include/stdlib/math/base/special/acsch.h:30]:
```suggestion
* Computes the hyperbolic arccosecant of a number.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/lib/native.js [hunk 18]
```diff
@@ -0,0 +1,62 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/lib/native.js:4]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/src/Makefile [hunk 19]
```diff
@@ -0,0 +1,70 @@
+#/
+# @license Apache-2.0
+#
+# Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/src/Makefile:4]:
```suggestion
# Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/src/addon.c [hunk 20]
```diff
@@ -0,0 +1,23 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/src/addon.c:4]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/src/main.c [hunk 21]
```diff
@@ -0,0 +1,34 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2022 The Stdlib Authors.
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/src/main.c:4]:
```suggestion
* Copyright (c) 2024 The Stdlib Authors.
```

File: lib/node_modules/@stdlib/math/base/special/acsch/src/main.c [hunk 22]
```diff
@@ -0,0 +1,34 @@
...
+
+/**
+* Computes the hyperbolic arccosecant of a number.
+*
+* @param x    input value
+* @return	  output value
+*
+* @example
+* out = stdlib_base_acsch( 2.0 );
+* // returns ~0.5493
```
Comment (Pranavchiku) [inline lib/node_modules/@stdlib/math/base/special/acsch/src/main.c:30]:
```suggestion
* out = stdlib_base_acsch( 1.0 );
* // returns ~0.881
```

**Review by Pranavchiku — CHANGES_REQUESTED** [review]
Thanks for this PR! Requested changes.

File: lib/node_modules/@stdlib/math/base/special/acsch/benchmark/c/benchmark.c (hunk 5 above)
Comment (yaswanthkosuru) [reply lib/node_modules/@stdlib/math/base/special/acsch/benchmark/c/benchmark.c:103]: ok thank u for pointing in right direction it should be x = ( 200.0*rand_double() ) -100(referenced from benchmark.js in same file)
